benchmarks/perf-test/src/compare_threading.rs

```diff
@@ -0,0 +1,181 @@
+use cervo_core::prelude::Inferer;
+use cervo_core::prelude::Response;
+use std::collections::HashMap;
+use cervo_runtime::BrainId;
+use cervo_runtime::AgentId;
+use cervo_runtime::Runtime;
+use std::time::Duration;
+use std::time::Instant;
+
+/// Given an existing runtime with brains, push new tickets based on inputs.
+fn push_tickets(runtime: &mut Runtime, batch_size: usize) {
+    let mut observation_vec = vec![];
+    for (inputs, brain_id) in runtime.models.iter().map(|model| {
+        (model.inferer.input_shapes().to_vec(), model.id)
+    })
+    {
+        let observations = crate::helpers::build_inputs_from_desc(batch_size as u64, &inputs);
+        for (key, val) in observations.iter() {
+            observation_vec.push((brain_id.clone(), &key, val.clone()));
+        }
+    }
+    for (brain_id, key, val) in observation_vec {
+        runtime
+        // TODO: LUc: Entrypoint:
+            .push(brain_id, *key.clone(), val)
+            .expect(&format!(
+                "Could not push to runtime key: {}, val: {:?}",
+                key, val
+            ));
+    }
+}
+
+
+/// Create a new runtime with for `brain_repetitions` times the brains in `onnx_paths`.
+/// Also generates observations based on the inferers' input shapes.
+fn add_inferers_to_runtime(
+    runtime: &mut Runtime,
+    onnx_paths: &[&str],
+    brain_repetitions: usize,
+    batch_size: usize,
+) {
+    for _ in 0..brain_repetitions {
+        for onnx_path in onnx_paths {
+            let mut reader = crate::helpers::get_file(onnx_path).expect("Could not open file");
+            let inferer = cervo_onnx::builder(&mut reader)
+                .build_fixed(&[batch_size])
+                .unwrap();
+
+            let inputs = inferer.input_shapes().to_vec();
+            let observations = crate::helpers::build_inputs_from_desc(batch_size as u64, &inputs);
+            let brain_id = runtime.add_inferer(inferer);
```

Comment:
@jaxs-ribs Store the brain ids and input shapes here and return them, then you don't need push_tickets.

Reply:
(I think you can query the input API with `Runtime::input_shapes`, so you technically only need to store the brain keys.)